src/alda/Util.java

```diff
@@ -43,6 +45,32 @@
     return concat(a, new Object[]{b});
   }
 
+  // Given an array of choices like {"yes", "no", "quit"}, offers them to the
+  // user as choices in this format: (y)es, (n)o, (q)uit
+  //
+  // Returns the selected string as soon as the user presses the corresponding
+  // key.
+  public static String promptWithChoices(List<String> choices)
+  throws IOException {
+    Map<Character, String> m = new LinkedHashMap<Character, String>();
+    for (String choice : choices) {
+      m.put(choice.charAt(0), choice);
+    }
+
+    String choiceString = choices.stream()
+                                 .map(str -> "(" + str.charAt(0) + ")"
+                                             + str.substring(1))
+                                 .collect(Collectors.joining(", "));
+
+    System.out.println(choiceString);
+
+    while (true) {
+      char c = (char) System.in.read();
+      String str = m.get(c);
```

Comment:
We probably want to downcase `c` (and insert lowercase into the hashmap above), so that `Y` works if `y` works.